realm/realm-library/src/androidTest/java/io/realm/RealmIntegerTests.java

```diff
@@ -59,228 +63,318 @@ public void tearDown() throws Exception {
+        ri.increment(17);
+        ri.decrement(7);
+        realm.commitTransaction();
+
+        assertEquals(Long.valueOf(42L), unmanagedRI.get());
+        assertEquals(Long.valueOf(47L), managedRI.get());
+    }
+
+    /**
+     * Assure that a MutableRealmInteger acquired from an unmanaged object is not affected by changes in the DB.
+     */
+    @Ignore("Not yet wired to core")
+    @Test
+    public void copyFromIsNotLive() {
+        realm.beginTransaction();
+        realm.createObject(Counters.class).getColumnCounter().set(42L);
+        realm.commitTransaction();
+
+        Counters obj = realm.where(Counters.class).findFirst();
+        MutableRealmInteger managedRI = obj.getColumnCounter();
+        MutableRealmInteger unmanagedRI = realm.copyFromRealm(obj).getColumnCounter();
+
+        realm.beginTransaction();
+        MutableRealmInteger ri = realm.where(Counters.class).findFirst().getColumnCounter();
+        ri.set(37);
+        ri.increment(17);
+        ri.decrement(7);
+        realm.commitTransaction();
+
+        assertEquals(Long.valueOf(42L), unmanagedRI.get());
+        assertEquals(Long.valueOf(47L), managedRI.get());
+    }
+
+    private void checkTransactionException(IllegalStateException e) {
+        assertTrue(e.getMessage().contains("only be done from inside a transaction"));
+    }
+
+    @SuppressWarnings({"ReferenceEquality", "EqualsIncompatibleType"})
+    private void testBasic(MutableRealmInteger r1, MutableRealmInteger r2) {
+        assertFalse(r1 == r2);
+
+        r1.set(10);
+        r2.set(Long.valueOf(10));
+        assertEquals(r1, r2);
+
+        r1.set(15);
+        r1.decrement(2);
+        r2.increment(3);
+        assertEquals(r1, r2);
+
+        MutableRealmInteger r3 = r1;
+        r1.set(19);
+        assertEquals(19, r3.get().intValue());
+    }
+
+    @SuppressWarnings({"ReferenceEquality", "EqualsIncompatibleType"})
+    private void testEquality(Counters c1, Counters c2) {
+        c1.columnCounter.set(7);
+        c2.columnCounter.set(Long.valueOf(7));
+        assertTrue(c1.columnCounter != c2.columnCounter);
+        assertTrue(c1.columnCounter.equals(c2.columnCounter));
+
+        MutableRealmInteger r1 = c1.columnCounter;
+        r1.increment(1);
+        assertTrue(r1.equals(c1.columnCounter));
+        assertTrue(r1 == c1.columnCounter);
+        assertTrue(c1.columnCounter.get().equals(8L));
+        assertFalse(c1.columnCounter.equals(c2.columnCounter.get()));
+        assertTrue(c1.columnCounter.get().intValue() == 8);
+
+        Long n = c1.columnCounter.get();
+        assertTrue(n.equals(Long.valueOf(8)));
+        assertTrue(n.equals(c1.columnCounter.get()));
+        assertTrue(n.intValue() == c1.columnCounter.get().intValue());
+
+        c1.columnCounter.increment(1);
+        assertFalse(n.intValue() == c1.columnCounter.get().intValue());
+        assertFalse(n.intValue() == r1.get().intValue());
+    }
+
+    private void testNullability(Counters c1) {
```

Comment:
I can do this if you like, but I think it is a horrible idea.  As you well know, defining the semantics for this feature has been a long and painful discussion, often plagued by misunderstanding.  The idea that English which is never checked for accuracy, better represents the semantic definition, than the code, seems completely mistaken, to me.